custom_components/omnik_inverter/config_flow.py

```diff
@@ -141,6 +144,46 @@ async def async_step_setup_html(
             errors=errors,
         )
 
+    async def async_step_setup_tcp(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle setup flow for tcp route."""
+        errors = {}
+
+        if user_input is not None:
+            try:
+                async with OmnikInverter(
+                    host=user_input[CONF_HOST],
+                    source_type=self.source_type,
+                    serial_number=user_input[CONF_SERIAL],
+                ) as client:
+                    await client.inverter()
+            except OmnikInverterError:
+                errors["base"] = "cannot_connect"
+            else:
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME],
+                    data={
+                        CONF_HOST: user_input[CONF_HOST],
+                        CONF_SOURCE_TYPE: self.source_type,
+                        CONF_SERIAL: user_input[CONF_SERIAL],
+                    },
+                )
+
+        return self.async_show_form(
+            step_id="setup_tcp",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_NAME, default=self.hass.config.location_name
+                    ): str,
+                    vol.Required(CONF_HOST): str,
+                    vol.Required(CONF_SERIAL): int,
```

Comment:
`SCAN_INTERVAL` uses `vol.All(vol.Coerce(int), vol.Range(min=1))`, would that be better as it allows freeform strings? Haven't figured out a minimum sensible value for this S/N though.

Reply:
why?

Reply:
Having a numeric field for a serial number seems slightly odd. Sure, it must be a string consisting of purely digits, but the "increment/decrement" buttons feel out of place. I had assumed `vol.All` would turn this config option into a text input, which would subsequently be validated to cast/coerce to a numeric input (ie. any non-digit character is invalid) but that's not the case leaving this moot.